polyfill/test/validStrings.mjs

```diff
@@ -0,0 +1,340 @@
+// Generates 1000 valid ISO 8601 strings according to the EBNF grammar and
```

Comment:
I didn't review this at all.

Reply:
Wow, this test is quite a handful. I'd have to block some time out to understand it too. @ptomato I'm curious: why this as opposed to just verifying a bunch of hand-picked strings? We could make sure to include all edge-cases in the test cases...

Reply:
Don't worry, it doesn't run as part of the automated test suite. You can only run it manually. It's meant more as a development tool.

I found it very useful while developing the grammar and the regex, because on the one hand you can test and debug the grammar by inputting it in this EBNF-like framework and examining the "valid" strings it outputs to see if they are really valid. Once you are happy with the grammar, you can then feed the strings into from() and investigate when it breaks. Anything that I found to be worth testing for, I also added manually to the test cases. With this tool, both the grammar and the regex ended up better than I would have been able to do without it.

I would only expect this file to be used when making a change to either the grammar or the regex. It doesn't necessarily need to be checked in to the repo, but my reasoning was, better to check it in than leave it only on my machine.